api.py

```diff
@@ -0,0 +1,56 @@
+import requests
+import json
+import urllib.parse
+from dotenv import dotenv_values
+
+class ApiController(object):
+    def __init__(self, url: str):
+        self.url = url if url[-1] != '/' else url[:-1]
+        self.endpoints = {
+            'levels': '/levels',
+            'groups': '/groups',
+            'schedules': '/schedules'
+        }
+
+    def get_levels(self):
+        return requests.get(self.url + self.endpoints['levels']).json()['response']
+
+    def get_groups(self, level_id: str | int):
+        params = f'?level_id={level_id}'
+        return requests.get(self.url + self.endpoints['groups'] + params).json()['response']
```

Comment:
Надо бы во все методы добавить проверку на ошибку в ответе

```suggestion
        response = requests.get(self.url + self.endpoints['groups'] + params).json()
        status = 'success' if 'response' in response else 'error'
        response = response['response'] if status == 'success' else response['error']
        return {'status': status, 'response': response}
```